### ISSUE TITLE
Respect scipen option in base

### ISSUE THREAD

**Issue by sumtxt** [r-lib/pillar#110]
E.g. if options(scipen=999) not use exponential notation.

**Comment by randomgambit** [comment]
hello there! is there a way to disable scipen in tibble (even manually)? thanks!

**Comment by krlmlr** [comment]
Would you like to test-drive `num()`, available in pillar 1.5.1.9002, to be released as pillar 1.6.0? After release, I'll reexport this the constructor from tibble too. See https://pillar.r-lib.org/dev/articles/numbers.html for a documentation stub.

``` r
library(pillar)
library(tibble)

x <- c(987543123578890.23, 240234.131, 40234.1)
col_a <- num(x, notation = "dec")

col_a
#> <tibble_num(dec)[3]>
#> [1] 987543123578890.          240234.           40234.
tibble::tibble(x, col_a)
#> # A tibble: 3 x 2
#>         x            col_a
#>     <dbl>            <dec>
#> 1 9.88e14 987543123578890.
#> 2 2.40e 5          240234.
#> 3 4.02e 4           40234.
```

<sup>Created on 2021-04-03 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

**Comment by krlmlr** [comment]
I've decided against querying the `"scipen"` option, instead you can set the new `"pillar.max_dec_width"` option which defaults to 13. See https://pillar.r-lib.org/dev/articles/digits.html for details.
